lib/puppet.rb

```diff
@@ -251,9 +251,11 @@ def self.base_context(settings)
       # doesn't exist
       default_environment = Puppet[:environment].to_sym
       if default_environment == :production
+        modulepath = settings[:modulepath]
+        modulepath = modulepath.nil? ? basemodulepath : Puppet::Node::Environment.split_path(modulepath)
         loaders << Puppet::Environments::StaticPrivate.new(
           Puppet::Node::Environment.create(default_environment,
-                                           basemodulepath,
+                                           modulepath,
```

Comment:
Should this be `modulepath + basemodulepath`, as opposed to overriding it completely? I'm not sure if that's the right thing...